### PR TITLE
feat(account): add social links

### PR DESCRIPTION
**Pull request by sthelemann**
The tables social_network and account_social_link have been added to accommodate an account owner's user names in social networks.

### CODE REVIEW

File: src/deploy/table_social_network.sql [hunk 1]
```diff
@@ -0,0 +1,20 @@
+-- Deploy maevsi:table_social_network to pg
+-- requires: schema_public
+
+BEGIN;
+
+CREATE TABLE maevsi.social_network (
+  name TEXT PRIMARY KEY
+);
```
Comment (dargmuesli) [inline src/deploy/table_social_network.sql:8]: Do you see upcoming extensions of this information or would an enumeration better fit this more or less fixed set of elements?
If we expect non-developers, like when event organizers, moderators, ... to manage such entities (social networks), we should create this table, but I think maevsi does not intend to offer social network tweaking.

**Review by dargmuesli — COMMENTED** [review]
General structure looks good to me! :white_check_mark:

File: src/deploy/table_social_network.sql (hunk 1 above)
Comment (sthelemann) [reply src/deploy/table_social_network.sql:8]: Additional records to this table will happen rarely but I prefer tables like this to enum types.

**Review by dargmuesli — CHANGES_REQUESTED** [review]
Thank you for the PR! The git commit had a trailing dot which I removed for you. I'll also work in the requested changes. Please review my changes if time allows :raised_hands:

File: src/deploy/table_social_network.sql (hunk 1 above)
Comment (dargmuesli) [reply src/deploy/table_social_network.sql:8]: Ok, I want to keep the API interface reasonable, so I converted the table into an enumeration which only appears as a type in the GraphQL API.

**Review by dargmuesli — APPROVED** [review]

File: src/sqitch.plan [hunk 2]
```diff
@@ -69,3 +69,6 @@ table_report [schema_public table_account_public table_event table_upload] 1970-
 table_report_policy [schema_public table_report role_account] 1970-01-01T00:00:00Z Marlon <marlon@maev.si> # Access policies for reports.
 table_legal_term [schema_public role_account role_anonymous] 1970-01-01T00:00:00Z Jonas Thelemann <e-mail+maevsi/sqitch@jonas-thelemann.de> # Legal terms like privacy policies or terms of service.
 table_legal_term_acceptance [schema_public table_account_public table_legal_term role_account] 1970-01-01T00:00:00Z Jonas Thelemann <e-mail+maevsi/sqitch@jonas-thelemann.de> # Tracks each user account's acceptance of legal terms and conditions.
+enum_social_network [schema_public] 2024-11-21T12:15:00Z sven <sven.thelemann@t-online.de> # Social networks.
+table_account_social_network [schema_public table_account_public enum_social_network] 2024-11-21T12:15:00Z sven <sven.thelemann@t-online.de> # Collects the account''s user names in social networks.
```
Comment (dargmuesli) [inline src/sqitch.plan:73]:
```suggestion
table_account_social_network [schema_public table_account_public enum_social_network] 2024-11-21T12:15:00Z sven <sven.thelemann@t-online.de> # Collects the account's user names in social networks.
```

Comment (dargmuesli) [reply src/sqitch.plan]: Done

File: src/deploy/table_account_social_link_policy.sql [hunk 3]
```diff
@@ -0,0 +1,22 @@
+-- Deploy maevsi:table_account_social_link_policy to pg
```
[the remark this answers is not in the document]
Comment (dargmuesli) [reply src/deploy/table_account_social_link_policy.sql]: Done

File: src/sqitch.plan [hunk 4]
```diff
@@ -69,3 +69,6 @@ table_report [schema_public table_account_public table_event table_upload] 1970-
 table_report_policy [schema_public table_report role_account] 1970-01-01T00:00:00Z Marlon <marlon@maev.si> # Access policies for reports.
 table_legal_term [schema_public role_account role_anonymous] 1970-01-01T00:00:00Z Jonas Thelemann <e-mail+maevsi/sqitch@jonas-thelemann.de> # Legal terms like privacy policies or terms of service.
 table_legal_term_acceptance [schema_public table_account_public table_legal_term role_account] 1970-01-01T00:00:00Z Jonas Thelemann <e-mail+maevsi/sqitch@jonas-thelemann.de> # Tracks each user account's acceptance of legal terms and conditions.
+table_social_network [schema_public] 2024-11-21T12:15:00Z sven <sven.thelemann@t-online.de> # Social networks.
+table_account_social_link [schema_public table_account_public table_social_network] 2024-11-21T12:15:00Z sven <sven.thelemann@t-online.de> # Collects the account''s user names in social networks.
+table_account_social_link_policy [schema_public table_social_network role_account] 2024-11-21T12:15:00Z sven <sven.thelemann@t-online.de> # Row level security policies for table account_social_link.
```
[the remark this answers is not in the document]
Comment (dargmuesli) [reply src/sqitch.plan]: Done

File: src/deploy/table_account_social_link_policy.sql [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+BEGIN;
+
+GRANT SELECT, INSERT, DELETE ON TABLE maevsi.account_social_link TO maevsi_account;
+
+-- Only allow selects for social links of the current account.
+CREATE POLICY account_social_link_select ON maevsi.account_social_link FOR SELECT USING (
+  account_id = NULLIF(current_setting('jwt.claims.account_id', true), '')::UUID
+);
```
[the remark this answers is not in the document]
Comment (dargmuesli) [reply src/deploy/table_account_social_link_policy.sql]: Done

File: src/deploy/table_account_social_link_policy.sql [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+-- Only allow inserting social links of the current account.
+CREATE POLICY account_social_link_insert ON maevsi.account_social_link FOR INSERT WITH CHECK (
+  account_id = NULLIF(current_setting('jwt.claims.account_id', true), '')::UUID
+);
+
+-- Only allow deleting social links of the current account..
```
[the remark this answers is not in the document]
Comment (dargmuesli) [reply src/deploy/table_account_social_link_policy.sql]: Done

**Comment by dargmuesli** [comment]
@sthelemann also check out the tests I've [added to this PR](https://github.com/maevsi/sqitch/blob/761d025fd9808257004cf8e461afc7818f0987ee/src/verify/table_account_social_network_policy.sql)!

**Pull request merged**